### PR TITLE
feat(scm): support scm trigger

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/handler/v1alpha1/workflowtrigger.go [hunk 1]
```diff
@@ -69,6 +78,38 @@ func UpdateWorkflowTrigger(ctx context.Context, project, workflowtrigger, tenant
...
+		// Handle trigger type change and repo change when SCM type.
+		// Do not care about the change of secret.
+		oldSpec := origin.Spec
+		newSpec := wft.Spec
+		unregisterOld, registerNew := false, false
+		if oldSpec.Type == v1alpha1.TriggerTypeSCM {
+			// Need to unregister old SCM webhook only when:
+			// * new trigger is not SCM type
+			// * repo of new trigger is different from old
+			if newSpec.Type == v1alpha1.TriggerTypeCron {
```
Comment (zhujian7) [inline pkg/server/handler/v1alpha1/workflowtrigger.go]: `if newSpec.Type != v1alpha1.TriggerTypeSCM { ` is better?

File: pkg/server/handler/v1alpha1/workflowtrigger.go [hunk 2]
```diff
@@ -82,7 +123,162 @@ func UpdateWorkflowTrigger(ctx context.Context, project, workflowtrigger, tenant
...
 }
+
+func registerSCMWebhook(tenant, wftName, secretName, repo string) error {
+	secret, err := handler.K8sClient.CoreV1().Secrets(common.TenantNamespace(tenant)).Get(
+		common.IntegrationSecret(secretName), metav1.GetOptions{})
+	if err != nil {
+		return cerr.ConvertK8sError(err)
+	}
+
+	repos := map[string][]string{}
```
Comment (zhujian7) [inline pkg/server/handler/v1alpha1/workflowtrigger.go]: need a comment to explain what the key/value represents.

File: pkg/server/handler/v1alpha1/workflowtrigger.go [hunk 3]
```diff
@@ -82,7 +123,162 @@ func UpdateWorkflowTrigger(ctx context.Context, project, workflowtrigger, tenant
...
+	}
+
+	if !found {
+		log.Warningf("Not found wft %s for repo %s in secret %s", wftName, repo, secretName)
+		return nil
+	}
+
+	reposStr, err := json.Marshal(repos)
+	if err != nil {
+		log.Errorf("Failed to marshal repos for secret")
```
Comment (zhujian7) [inline pkg/server/handler/v1alpha1/workflowtrigger.go]: which secret?

**Comment by supereagle** [comment]
@zhujian7  Have fixed comments, PTAL

**Pull request merged**